cpp/arcticdb/version/version_map_batch_methods.hpp

```diff
@@ -117,6 +120,130 @@ inline std::shared_ptr<std::unordered_map<std::pair<StreamId, VersionId>, AtomKe
     return output;
 }
 
+struct StreamVersionData {
```

Comment:
@vasil-pashov moving the comment here https://github.com/man-group/ArcticDB/pull/423/files#r1205054793

Reply:
However, it is a struct. I think it is a good practice to define it in the app file

Reply:
It's an implementation detail, it belongs with the implementation. I don't think we should declare structs in the header file as a matter of course, when they are not part of the API that we want to expose

Reply:
To summarise:
We don’t want to define/declare this new struct in separate hpp/cpp files because it would give the wrong idea to the users (i.e. it can be used elsewhere). On the other side, even if the functions were defined in the cpp, we woudln’t want to define this struct in the cpp, otherwise the tests would not have access to its definition. It is also possible to declare a struct in the hpp and define in the cpp, but it doesn’t seem to be a common practice

Reply:
Let's move the definition of `StreamVersionData` to it's own header file (and possibly it's own cpp file).

Reply:
We will do it in another PR, as has been commonly decided https://github.com/man-group/ArcticDB/issues/452